internal/states/instance_object.go

```diff
@@ -94,7 +117,7 @@ const (
 // The returned object may share internal references with the receiver and
 // so the caller must not mutate the receiver any further once once this
 // method is called.
-func (o *ResourceInstanceObject) Encode(ty cty.Type, schemaVersion uint64) (*ResourceInstanceObjectSrc, error) {
+func (o *ResourceInstanceObject) Encode(schema providers.Schema) (*ResourceInstanceObjectSrc, error) {
```

Comment:
something to come back to later might be to update the comments for the functions whose signature have changed

Reply:
Actually, we should probably update them along with the change, or else we'll never find them again ;)